packages/syft/src/syft/core/tensor/tensor.py

```diff
@@ -3,31 +3,165 @@
 
 # stdlib
 from typing import Any
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
 
 # third party
-from google.protobuf.reflection import GeneratedProtocolMessageType
 import numpy as np
 import torch as th
 
 # relative
-from ...core.common.serde.serializable import Serializable
-from ...proto.core.tensor.tensor_pb2 import Tensor as Tensor_PB
-from ..common.serde.deserialize import _deserialize as deserialize
+from ... import lib
+from ...ast.klass import pointerize_args_and_kwargs
+from ...core.common.serde.recursive import RecursiveSerde
+from ...util import inherit_tags
 from ..common.serde.serializable import bind_protobuf
-from ..common.serde.serialize import _serialize as serialize
+from ..common.uid import UID
+from ..node.abstract.node import AbstractNodeClient
+from ..node.common.action.run_class_method_action import RunClassMethodAction
+from ..pointer.pointer import Pointer
+from .ancestors import AutogradTensorAncestor
+from .ancestors import PhiTensorAncestor
 from .fixed_precision_tensor_ancestor import FixedPrecisionTensorAncestor
 from .passthrough import PassthroughTensor  # type: ignore
-from .smpc.mpc_tensor_ancestor import MPCTensorAncestor
 
+# from .autodp.single_entity_phi import SingleEntityPhiTensor
+# from .autodp.single_entity_phi import SingleEntityPhiTensorPointer
+from .smpc.mpc_tensor import MPCTensor
 
+
+class TensorPointer(Pointer):
+
+    # Must set these at class init time despite
```

Comment:
Refactor?

Reply:
Nah I think this is how it should work for now unless you're referring to some refactor i'm not aware of.